src/Bicep.Core/Emit/TemplateWriter.cs

```diff
@@ -987,18 +994,26 @@ private void EmitVariablesIfPresent(ExpressionEmitter emitter, IEnumerable<Decla
             });
         }
 
-        private void EmitProviders(ExpressionEmitter emitter, ImmutableArray<DeclaredProviderExpression> providers)
+        private void EmitExtensionsIfPresent(ExpressionEmitter emitter, ImmutableArray<DeclaredProviderExpression> providers)
         {
             if (!providers.Any())
             {
                 return;
             }
 
+            // TODO: Remove if statement once all providers got migrated to extensions (extensibility v2 contract).
             if (this.Context.SemanticModel.Features.LocalDeployEnabled)
```

Comment:
For consistency, could we either access this property with `Context.Settings.LocalDeployEnabled` everywhere (the convenience property you added), or with `Context.SemanticModel.Features.LocalDeployEnabled` everywhere?

Reply:
Got it, I didn't know you get access to Settings or SemanticModel from Context. I'll remove the convenience property and use `Context.SemanticModel.Features.LocalDeployEnabled` everywhere instead.